solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -1222,11 +1222,22 @@ public boolean isDirectUpdatesToLeadersOnly() {
   /**
    * If caches are expired they are refreshed after acquiring a lock. use this to set the number of
    * locks
+   *
+   * @deprecated use {@link CloudHttp2SolrClient.Builder#setParallelCacheRefreshes(int)} instead
    */
+  @Deprecated
   public void setParallelCacheRefreshes(int n) {
     locks = objectList(n);
   }
 
+  /**
+   * If caches are expired they are refreshed after acquiring a lock. This method sets the number of
+   * locks. It is used by the Builder only.
+   */
+  void setParallelCacheRefreshesLocks(int parallelCacheRefreshesLocks) {
+    locks = objectList(parallelCacheRefreshesLocks);
+  }
+
```

Comment:
was looking for a place to put this observation, so I'll leave it here. I think I would prefer having a method that takes an 'int' and hides away the implementation. this would allow in the future for replacing this cache striping with another cache that also allows for setting the stripe size without breaking any clients that rely on this (if ever).